provision.sh

```diff
@@ -288,14 +315,51 @@ function provision {
         fi
     fi
 
-    printf "Setting LNS address to ${stackAddr}...\n" >&2
-    setLorafwdKeyQuoted "${sshCmd}" "node" "${stackAddr}"
+    pushConfig  "${gatewayAddr}" \
+                "${apiKey}" \
+                "${stackAddr}/api/v3/gcs/gateways/${gatewayID}/cpf/lorad/lorad.json" \
+                "${loradDir}/etc/lorad.json"
+
+    pushConfig  "${gatewayAddr}" \
+                "${apiKey}" \
+                "${stackAddr}/api/v3/gcs/gateways/${gatewayID}/cpf/lorafwd/lorafwd.toml" \
+                "${lorafwdDir}/etc/lorafwd.toml"
 
-    printf "Setting LNS uplink port to ${uplinkPort}...\n" >&2
-    setLorafwdKey "${sshCmd}" "service.uplink" "${uplinkPort}"
+    read -r -n 1 -p "Provisioning successfully finished, restart packet forwader?[y/n]" ans
+    printf '\n' >&2
+    if [[ ! "${ans}" = "y" ]]; then
+        printf "Please manually restart the packet forwarder by executing \`ssh root@${gatewayAddr} /etc/init.d/knet restart\`\n" >&2
+        return 0
+    fi
+    printf "Restarting packet forwarder...\n" >&2
+    ${sshCmd} "/etc/init.d/knet restart"
+    return 0
+}
 
-    printf "Setting LNS downlink port to ${downlinkPort}...\n" >&2
-    setLorafwdKey "${sshCmd}" "service.downlink" "${downlinkPort}"
+function printUsage {
+    printf "Usage: $0 GATEWAY-ADDRESS STACK-ADDRESS GATEWAY-ID API-KEY\n" >&2
```

Comment:
nit; `<gateway-address> <stack-address> ...`

Reply:
This is very much inconsistent across different tools:
![2019-12-20-18:04:52-screenshot](https://user-images.githubusercontent.com/12877905/71271025-45e33f80-2353-11ea-87d4-7bfce08a6850.png)
![2019-12-20-18:02:36-screenshot](https://user-images.githubusercontent.com/12877905/71270967-26e4ad80-2353-11ea-8d09-e52ead92ec67.png)
![2019-12-20-18:02:48-screenshot](https://user-images.githubusercontent.com/12877905/71270975-2b10cb00-2353-11ea-8314-5be78e98dd8b.png)

But in general, from what I have seen, usually:
- `[]` enclose optional parameters
- no `[]` means that parameters are mandatory
- `<>` is only used by Go

I used the suggested format, because most "old-school" Unix tooling uses that.
I don't really have a stong opinion about this, however I think `<>` may be confusing, since that syntax is very uncommon(i.e. the only tool I know of using that syntax is Go compiler)

Reply:
Yeah OK I agree. In the `ssh` and `grep` cases, the command is mandatory and in `[ ]`, though. If you execute them without a command, they just print help.

So maybe `[gateway_address] [stack_address] ...`?

Reply:
The only mandatory part of `ssh` is `destination`, which is without `[]`
In case of `grep`, the only mandatory part is the `pattern`, which is, again, without `[]`
Note that `FILE` is not mandatory for `grep`:
```
$ echo "asd" | grep     
Usage: grep [OPTION]... PATTERNS [FILE]...
Try 'grep --help' for more information.
$ echo "asd" | grep "a"
asd
```